File: codeTemplate/entityDataAccess.java

	static public void createTable()
	{
		SqlSession session = Common.getSession();
		DataMapper mapper = session.getMapper(DataMapper.class);
		try {
			mapper.createTable@entity@();
			session.commit();
		} catch (Exception e) {
			e.printStackTrace();

		} finally {
			session.close();
		}
	}
	
	static public @entity@ get(int id)
	{
		SqlSession session = Common.getSession();
		@entity@ result = null;
		DataMapper mapper = session.getMapper(DataMapper.class);
		try {
			result = mapper.get@entity@(id);
			session.commit();
		} catch (Exception e) {
			e.printStackTrace();

		} finally {
			session.close();
		}
		
		return result;
	}
	

	static public void delete(@entity@ one)
	{
		SqlSession session = Common.getSession();
		DataMapper mapper = session.getMapper(DataMapper.class);
		try {
			mapper.delete@entity@(one);
			session.commit();
		} catch (Exception e) {
			e.printStackTrace();

		} finally {
			session.close();
		}
		
	}
	

	static public void update(@entity@ one)
	{
		SqlSession session = Common.getSession();
		DataMapper mapper = session.getMapper(DataMapper.class);
		try {
			mapper.update@entity@(one);
			session.commit();
		} catch (Exception e) {
			e.printStackTrace();

		} finally {
			session.close();
		}
	}


	static public void insert(@entity@ one)
	{
		SqlSession session = Common.getSession();
		DataMapper mapper = session.getMapper(DataMapper.class);
		try {
			mapper.insert@entity@(one);
			int id = mapper.getLastID();
			one.setId(id);
			session.commit();
		} catch (Exception e) {
			e.printStackTrace();

		} finally {
			session.close();
		}
	}
	



	static public int count(String condition)
	{
		SqlSession session = Common.getSession();
		int result = 0;
		DataMapper mapper = session.getMapper(DataMapper.class);
		try {
			result = mapper.count@entity@(condition);
			session.commit();
		} catch (Exception e) {
			e.printStackTrace();

		} finally {
			session.close();
		}
		
		return result;
	}
	
	

File: codeTemplate/entityDataAccess.cs


		public static @entity@ Find(@entity@ obj)
		{
			if (obj != null)
			{
				const string strKey = "@entity@.Find";
				try
				{
					return MapperInstance.SqlMapper.QueryForObject<@entity@>(strKey, obj);
				}
				catch (Exception e)
				{
					string strStack = e.StackTrace;
					string strMsg = e.Message;
				}
			}
			return null;
		}

		public static void Insert(@entity@ obj)
		{
			if (obj != null)
			{
				const string strId = "@entity@.Insert";
				const string strId2 = "@entity@.LastID";
				try
				{
					MapperInstance.SqlMapper.Insert(strId, obj);

					obj.id = MapperInstance.SqlMapper.QueryForObject<NumBase>(strId2, obj).Num;
				}
				catch (Exception e)
				{
					string strStack = e.StackTrace;
					string strMsg = e.Message;
				}
			}
		}

		public static void Update(@entity@ obj)
		{
			if (obj != null)
			{
				const string strId = "@entity@.Update";
				try
				{
					MapperInstance.SqlMapper.Update(strId, obj);
				}
				catch (Exception e)
				{
					string strStack = e.StackTrace;
					string strMsg = e.Message;
				}
			}
		}

		public static void Delete(@entity@ obj)
		{
			if (obj != null)
			{
				const string strId = "@entity@.Delete";
				try
				{
					MapperInstance.SqlMapper.Delete(strId, obj);
				}
				catch (Exception e)
				{
					string strStack = e.StackTrace;
					string strMsg = e.Message;
				}
			}
		}

		public static IList<@entity@> FindList(string strPara)
		{
			const string strId = "@entity@.FindList";
			try
			{
				IList<@entity@> result = MapperInstance.SqlMapper.QueryForList<@entity@>(strId, strPara);
				return result;
			}
			catch (Exception e)
			{
				string strStack = e.StackTrace;
				string strMsg = e.Message;
				return null;
			}
		}


		public static int Count(string strPara)
		{
			const string strId = "@entity@.Count";
			try
			{
				var n = MapperInstance.SqlMapper.QueryForObject<NumBase>(strId, strPara);
				return n.Num;
			}
			catch (Exception e)
			{
				string strStack = e.StackTrace;
				string strMsg = e.Message;
				return 0;
			}
		}

